### PR TITLE
Made working directory accessible for plugins

### PR DESCRIPTION
**Pull request by Czompi**
### Made working directory accessible for plugins
**Summary:**
- Throw an `UnauthorizedAccessException` if the plugin uses full path that is outside plugin folder.
- Append working directory to relative paths (in `FileReaderService` & `FileWriterService` classes)
- Code cleanup and change `CreateWorkingDirectory(bool, bool)` method description
- Added an example `IFileReader.CreateWorkingDirectory(bool, bool)` usage to `SamplePlugin`
- Made `SamplePlugin`'s log more colorful

**Added methods**
- `IFileReader.CreateWorkingDirectory(bool, bool)`
- `IFileWriter.CreateWorkingDirectory(bool, bool)`
- `FileReaderService.CreateWorkingDirectory(bool, bool)`
- `FileWriterService.CreateWorkingDirectory(bool, bool)`

### CODE REVIEW

File: Obsidian.API/Plugins/Services/IFileReader.cs [hunk 1]
```diff
@@ -76,5 +76,13 @@ public string[] GetSubdirectories(string path)
...
             return Directory.GetDirectories(path);
         }
+
+        /// <summary>
+        /// Creates a directory, that is used by default for relative paths.
+        /// </summary>
+        /// <param name="createOwnDirectory">If set to <b><c>false</c></b>, the automatically assigned directory for your plugin will be skipped.</param>
+        /// <param name="skipFolderAutoGeneration">If set to <b><c>true</c></b>, skips the auto generation method for default plugin dir. Also, <b><c>createOwnDirectory</c></b> needs to be <b><c>true</c></b> for this to work.</param>
+        /// <returns>Path to the created directory.</returns>
+        public string CreateWorkingDirectory(bool createOwnDirectory = true, bool skipFolderAutoGeneration = false);
```
Comment (Seb-stian) [inline Obsidian.API/Plugins/Services/IFileReader.cs]: Could this definition (and the one in IFileWriter) be moved to IFileService?

**Review by Seb-stian — COMMENTED** [review]
Good job 👍

File: Obsidian.API/Plugins/Services/IFileReader.cs (hunk 1 above)
Comment (Czompi) [reply Obsidian.API/Plugins/Services/IFileReader.cs]: i think it can be moved. I'll check that.

Comment (Czompi) [reply Obsidian.API/Plugins/Services/IFileReader.cs]: moved

**Comment by Naamloos** [comment]
Feel free to ignore the appveyor build ;^)

**Comment by Naamloos** [comment]
LGTM

**Pull request merged**